hedera-mirror-rest/transactions.js

```diff
@@ -314,27 +350,45 @@ const getTransactionsInnerQuery = function (
       namedLimitQuery
     );
 
+    const namedNftTrAccountQuery = namedAccountQuery.replace(/ctl\.entity_id/g, 'ntl.sender_account_id');
+    const nftTrQuery = getTransferDistinctTimestampsQuery(
+      'nft_transfer',
+      'ntl',
+      namedTsQuery,
+      'consensus_timestamp',
+      resultTypeQuery,
+      transactionTypeQuery,
+      namedNftTrAccountQuery,
+      '',
+      order,
+      namedLimitQuery
+    );
+
     if (creditDebitQuery) {
       // credit/debit filter applies to crypto_transfer.amount and token_transfer.amount, a full outer join is needed to get
       // transactions that only have a crypto_transfer or a token_transfer
       return `
-        SELECT COALESCE(ctl.consensus_timestamp, ttl.consensus_timestamp) AS consensus_timestamp
+        SELECT COALESCE(ctl.consensus_timestamp, ttl.consensus_timestamp)
+                 AS consensus_timestamp
         FROM (${ctlQuery}) AS ctl
-        FULL OUTER JOIN (${ttlQuery}) as ttl
-        ON ctl.consensus_timestamp = ttl.consensus_timestamp
+               FULL OUTER JOIN (${ttlQuery}) as ttl
+                               ON ctl.consensus_timestamp = ttl.consensus_timestamp
         ORDER BY consensus_timestamp ${order}
-        ${namedLimitQuery}`;
+          ${namedLimitQuery}`;
     }
-    // account filter applies to transaction.payer_account_id, crypto_transfer.entity_id, and token_transfer.account_id, a full outer join
-    // between the three tables is needed to get rows that may only exist in one.
+    // account filter applies to transaction.payer_account_id, crypto_transfer.entity_id, and nft_transfer.account_id,
+    // and token_transfer.account_id, a full outer join between the three tables is needed to get rows that may only exist in one.
     return `
-        SELECT coalesce(t.consensus_timestamp,ctl.consensus_timestamp,ttl.consensus_timestamp) AS consensus_timestamp
-        FROM (${transactionOnlyQuery}) AS t
-        FULL OUTER JOIN (${ctlQuery}) AS ctl
-        ON t.consensus_timestamp = ctl.consensus_timestamp
-        FULL OUTER JOIN (${ttlQuery}) AS ttl
-        ON coalesce(t.consensus_timestamp, ctl.consensus_timestamp) = ttl.consensus_timestamp
-        ORDER BY consensus_timestamp ${order}
+      SELECT coalesce(t.consensus_timestamp, ctl.consensus_timestamp, ttl.consensus_timestamp,
+                      ntl.consensus_timestamp) AS consensus_timestamp
+      FROM (${transactionOnlyQuery}) AS t
+             FULL OUTER JOIN (${ctlQuery}) AS ctl
+                             ON t.consensus_timestamp = ctl.consensus_timestamp
+             FULL OUTER JOIN (${ttlQuery}) AS ttl
+                             ON coalesce(t.consensus_timestamp, ctl.consensus_timestamp) = ttl.consensus_timestamp
+             FULL OUTER JOIN (${nftTrQuery}) as ntl
+                             ON coalesce(t.consensus_timestamp, ctl.consensus_timestamp) = ntl.consensus_timestamp
```

Comment:
unfortunately have to coalesce the previous three timestamps:

```suggestion
                             ON coalesce(t.consensus_timestamp, ctl.consensus_timestamp, ttl.consensus_timestamp) = ntl.consensus_timestamp
```

should also update `transactions-35-` spec to have a nft transfer whose receiver account id is the account.id in the query filter to make sure we don't see duplicate nft trasnfers

Reply:
Updated but seem to be hitting the bug.

Reply:
Capturing this issue in [2139](https://github.com/hashgraph/hedera-mirror-node/issues/2139) in case we want to cut a tag and allow for release before I can get to resolving it